lib/TcpConnection.js

```diff
@@ -144,7 +144,7 @@ module.exports = function factory (userOptions) {
       }
     }
     if (options.reconnect) {
-      reconnect()
+      reconnect(hadError && socketError)
```

Comment:
@mcollina I'm thinking about it because it's potentially a breaking change but I wonder if we should reconnect if the socket was closed (or ended) without an error?

Reply:
no we shouldn't reconnect.

Reply:
done!
I've also added a test to make sure that a `close` event is sent on the stream when the socket is gracefully closed using `socket.destroy()` (without an error)